tests/test_common/test_managed_datetime.py

```diff
@@ -17,69 +18,147 @@
 
 
 class TestMgdDatetime(object):
+    """
+    Test the instantiation and usages of the managed datetime object.
+    """
 
-    @pytest.mark.parametrize('input, output', [(test_dt_instance, test_dt_instance),
-                                               (test_dt_instance.isoformat(), test_dt_instance),
-                                               (None, test_dt_instance_epoch)
-                                               ])
+    @pytest.mark.parametrize('dt_obj, nullable, expected_restriction', [
+        (datetime, False, R.DATETIME),
+        (datetime, True, R.NULL_DATETIME),
+        (date, False, R.DATE),
+        (date, True, R.NULL_DATE),
+        pytest.param(None, True, None, marks=pytest.mark.xfail(raises=AssertionError))
```

Comment:
Does this AssertionError come from the test's assertion or from the codes?

Reply:
The code